### PR TITLE
Use JsonTextReader in DependencyContextJsonReader 

### PR DESCRIPTION
**Pull request by pakrym**
This PR supersedes https://github.com/dotnet/cli/pull/3920

See aspnet/Mvc#5060

/cc @Yves57 @rynowak

Replace the high Json.Net concept of JObject.Load() by using the JsonTextReader class.

### CODE REVIEW

**Comment by Yves57** [comment]
@pakrym I was not too far from the solution :-).
Happy to have helped. And thank you for the fixes!

**Comment by eerhardt** [comment]
> using Newtonsoft.Json;

Copyright

---
Refers to: src/Microsoft.Extensions.DependencyModel/JsonTextReaderExtensions.cs:1 in 1e67538. [](commit_id = 1e67538115048f2432286b8b9d8a2cba587db03c, deletion_comment = False)

File: src/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs [hunk 1]
```diff
@@ -127,234 +121,533 @@ private DependencyContext Read(JObject root)
...
-        private static string[] ReadAssetList(JObject libraryObject, string name)
+        private List<RuntimeFallbacks> ReadRuntimes(JsonTextReader reader)
         {
-            var assembliesObject = (JObject)libraryObject[name];
+            var runtimeFallbacks = new List<RuntimeFallbacks>();
 
-            if (assembliesObject == null)
+            reader.ReadStartObject();
+
+            while (reader.Read() && reader.TokenType == JsonToken.PropertyName)
```
Comment (eerhardt) [inline src/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs]: Could use JsonTextReaderExtensions.TryReadStringProperty here.

File: src/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs [hunk 2]
```diff
@@ -127,234 +121,533 @@ private DependencyContext Read(JObject root)
...
-                yield break;
-            }
-            foreach (var libraryProperty in runtimeTargetObject)
+            string hash = null;
+            string type = null;
+            bool serviceable = false;
+
+            reader.ReadStartObject();
+
+            while (reader.Read() && reader.TokenType == JsonToken.PropertyName)
```
Comment (eerhardt) [inline src/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs]: Could use JsonTextReaderExtensions.TryReadStringProperty here.

File: src/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs [hunk 3]
```diff
@@ -127,234 +121,533 @@ private DependencyContext Read(JObject root)
...
+        private void ReadRuntimeTarget(JsonTextReader reader, out string runtimeTargetName, out string runtimeSignature)
+        {
+            runtimeTargetName = null;
+            runtimeSignature = null;
+
+            reader.Read();
+            if (reader.TokenType == JsonToken.String)
             {
-                yield return new RuntimeFallbacks(pair.Key, pair.Value.Values<string>().ToArray());
+                // This fallback is temporary
```
Comment (eerhardt) [inline src/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs]: How temporary is this fallback?  Should this comment be removed?  Should this check?

**Comment by eerhardt** [comment]
:shipit:

**Comment by pakrym** [comment]
@Yves57 thank you for your help!

**Pull request merged**